indy_node/test/nym_txn/test_nym_diddoc_content.py

```diff
@@ -0,0 +1,60 @@
+import copy
+import json
+from random import randint
+
+import pytest
+from indy_common.constants import NYM
+from indy_node.test.mock import build_nym_request
+from plenum.common.exceptions import RequestNackedException
+from plenum.test.helper import sdk_get_and_check_replies
+from plenum.test.pool_transactions.helper import sdk_sign_and_send_prepared_request
+
+
+def test_diddoc_content_added(
+    looper, sdk_pool_handle, sdk_wallet_endorser, diddoc_content
+):
+    _, did = sdk_wallet_endorser
+    nym_request = build_nym_request(did, did, None, diddoc_content, None)
+    request_couple = sdk_sign_and_send_prepared_request(
+        looper, sdk_wallet_endorser, sdk_pool_handle, nym_request
+    )
+    sdk_get_and_check_replies(looper, [request_couple])
+
+
+# Schema of NYM Tx in plenum is not enforced! Arbitrary properties can be injected, but won't be included in state
+def test_add_arbitrary_property(
+    looper, sdk_pool_handle, sdk_wallet_endorser
+):
+    _, did = sdk_wallet_endorser
+
+    nym_request = json.dumps(
+        {
+            "identifier": did,
+            "reqId": randint(100, 1000000),
+            "protocolVersion": 2,
+            "operation": {"dest": did, "type": NYM, "someProp": "someValue"},
+        }
+    )
+
+    request_couple = sdk_sign_and_send_prepared_request(
+        looper, sdk_wallet_endorser, sdk_pool_handle, nym_request
+    )
+    replies = sdk_get_and_check_replies(looper, [request_couple])
+
+    assert replies[0][1]["result"]["txn"]["data"]["someProp"] == "someValue"
+
+
+def test_add_didoc_with_id_fails(
```

Comment:
More thorough tests for doc validation are performed in `test_nym_handler`; should more tests be added here for diddoc behavior?

Reply:
Yes, like a test to ensure the data from an existing attrib (endpoint) is returned in the `diddocContent` when the associated nym is read.

Reply:
The reading of attrib data when diddocContent is absent is an optional legacy support feature implemented on the resolver side rather than on the ledger side.

Reply:
Ah, when I wrote this I was under the impression the assembly was happening ledger side.  I've re-read the spec and it definitely states this as being the responsibility of the client.  I think I originally tripped over the last paragraph in section 10.1.6, leaving myself believing the assembly was performed ledger side for some reason.